src/qr.rs

```diff
@@ -1024,6 +1037,101 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_decode_and_apply_dclogin() -> Result<()> {
+        let ctx = TestContext::new().await;
+
+        let result = check_qr(&ctx.ctx, "dclogin:usename+extension@host?p=1234&v=1").await?;
+        if let Qr::Login { address, options } = result {
+            assert_eq!(address, "usename+extension@host".to_owned());
+
+            if let LoginOptions::V1 { mail_pw, .. } = options {
+                assert_eq!(mail_pw, "1234".to_owned());
+            }
+        } else {
+            bail!("wrong type")
+        }
+
+        assert!(ctx.ctx.get_config(Config::Addr).await?.is_none());
```

Comment:
BTW, you don't necessarily need to write `ctx.ctx`, 
```suggestion
        assert!(ctx.get_config(Config::Addr).await?.is_none());
```
would also work

Reply:
let's do this in another pr:
`ctx.ctx.` is many times in the `qr.rs` file, even before this pr also 2 other files have it: `oauth2.rs` and `peerstate.rs`